streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java

```diff
@@ -118,8 +118,19 @@ private void putIfAbsent(final Properties props, final String propertyKey, final
      * Stop the Kafka cluster.
      */
     private void stop() {
+        if (brokers.length > 1) {
+            // delete the topics first to avoid cascading leader elections while shutting down the brokers
+            try {
+                deleteAllTopicsAndWait(60_000L);
+            } catch (final InterruptedException e) {
+                log.warn("Couldn't delete all topics before stopping brokers", e);
```

Comment:
Should we actually capture `AssertionError` instead of `InterruptedException`? I think when it times out the former is thrown not the latter.

Reply:
Oh, yeah, you're right. I was just reacting to the checked exception. I'll fix it. In fact, we should probably rethrow for the InterruptedException.